components/Mediator.vue

```diff
@@ -552,7 +556,68 @@ function updateHandlerWindow(handlerWindow) {
 
 async function getWebAppManifest(host) {
   try {
-    const response = await axios.get('/manifest', {params: {host}});
+    const url = `https://${host}/manifest.json`;
+    if(!cacheStorage && typeof caches !== 'undefined') {
+      cacheStorage = await caches.open(CACHE_NAME);
+    }
+    if(!cacheStorage) {
+      // no cache storage/API available, fetch directly
+      return getWebAppManifestFromUrl(url);
+    }
+
+    // try to get cached response
+    let response = await cacheStorage.match(url);
+    if(response && response.ok) {
+      const expires = new Date(response.headers.get('expires'));
+      const now = new Date();
+      if(expires >= now) {
+        // return cached response
+        return response.json();
+      }
+      // remove expired response from cache
+      await cacheStorage.delete(url);
+    }
+
+    try {
+      // fetch live response
+      response = await fetch(url);
```

Comment:
Should this be using `getWebAppManifestFromUrl`?  (and handling the possible null result appropriately)

Reply:
I see that api just returns data.  In any case, should this be using httpClient somehow?

Reply:
No, `httpClient` is only used in the degenerate case. I'd like to remove it entirely and just use fetch, but figured we may need to still support some old browser case.